core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -94,9 +111,109 @@ public final void setPort(final int port) {
+     *
+     * @return The host name.
+     */
+    public final String getProxyHost() {
+        return proxyHost;
+    }
+
+    /**
+     * Sets the name or literal IP address of the proxy host that the client should use to connect to the an AMQP host
+     *
+     * @param proxyHost The host name or IP address of the proxy
+     * @throws NullPointerException if proxyHost is {@code null}.
+     */
+    public final void setProxyHost(final String proxyHost) {
+        this.proxyHost = Objects.requireNonNull(proxyHost);
+    }
+
+    /**
+     * Gets the TCP port of the proxy server that this client is configured to use.
+     *
+     * @return The port number.
+     */
+    public final int getProxyPort() {
+        return proxyPort;
+    }
+
+    /**
+     * Sets the TCP port of the proxy server that this client should use.
+     *
+     * @param proxyPort The port number of the proxy
+     * @throws IllegalArgumentException if port &lt; 1024 or port &gt; 65535.
+     */
+    public final void setProxyPort(final int proxyPort) {
+        if (isInValidPortRange(proxyPort)) {
+            this.proxyPort = proxyPort;
+        } else {
+            throw new IllegalArgumentException("invalid proxy port number");
+        }
+    }
+
+    /**
+     * Gets the username of the proxy server the client is configured with.
+     *
+     * @return the username
+     */
+    public String getProxyUsername() {
+        return proxyUsername;
+    }
+
+    /**
+     * Sets the username to be used for authentication with the proxy server.
+     *
+     * @param proxyUsername username to be used for authentication with the proxy server
+     * @throws NullPointerException if proxyUsername is {@code null}.
+     */
+    public void setProxyUsername(String proxyUsername) {
+        this.proxyUsername = Objects.requireNonNull(proxyUsername);
+    }
+
+    /**
+     * Gets the password of the proxy user to be used in authentication.
+     *
+     * @return password in clear text.
+     */
+    public String getProxyPassword() {
+        return proxyPassword;
+    }
+
+    /**
+     * Sets the password of the proxy user to be used in authentication.
+     *
+     * @param proxyPassword password in clear text
+     * @throws NullPointerException if proxyPassword is {@code null}.
+     */
+    public void setProxyPassword(String proxyPassword) {
+        this.proxyPassword = Objects.requireNonNull(proxyPassword);
+    }
+
+    /**
+     * Gets the proxy type to be used. Default is {@link ProxyType#SOCKS5}.
+     *
+     * @return the proxy type if set by the user or the default {@link ProxyType#SOCKS5}.
+     */
+    public ProxyType getProxyType() {
+        return proxyType != null? proxyType : DEFAULT_PROXY_TYPE;
+    }
+
+    /**
+     * Sets the proxy type to be used.
+     *
+     * @param proxyType
+     * @throws IllegalArgumentException if the given proxyType is not a valid {@link ProxyType}
```

Comment:
refer to the list of supported values instead of linking to `ProxyType` which is not a parameter to this method.